recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,510 @@
-        'lto': ['On', 'Off', 'Full', 'Thin'],
-        'static_stdlib': [True, False],
-        'unwind_tables': [True, False],
-        'expensive_checks': [True, False],
-        'use_perf': [True, False],
-        'use_sanitizer': [
-            'Address',
-            'Memory',
-            'MemoryWithOrigins',
-            'Undefined',
-            'Thread',
-            'DataFlow',
-            'Address;Undefined',
-            'None'
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "components": ["ANY"],
+        "exceptions": [True, False],
+        "rtti": [True, False],
+        "threads": [True, False],
+        "lto": ["On", "Off", "Full", "Thin"],
+        "static_stdlib": [True, False],
+        "unwind_tables": [True, False],
+        "expensive_checks": [True, False],
+        "use_perf": [True, False],
+        "use_sanitizer": [
+            "Address",
+            "Memory",
+            "MemoryWithOrigins",
+            "Undefined",
+            "Thread",
+            "DataFlow",
+            "Address;Undefined",
+            "None"
         ],
-        'with_ffi': [True, False],
-        'with_zlib': [True, False],
-        'with_xml2': [True, False],
-        'use_llvm_cmake_files': [True, False],
+        "with_ffi": [True, False],
+        "with_libedit": [True, False],
+        "with_terminfo": [True, False],
+        "with_zlib": [True, False],
+        "with_xml2": [True, False],
+        "with_z3": [True, False],
+        "ram_per_compile_job": ["ANY"],
+        "ram_per_link_job": ["ANY"],
     }
+    options.update({f"with_target_{target.lower()}": [True, False] for target in LLVM_TARGETS})
     default_options = {
-        'shared': False,
-        'fPIC': True,
-        'components': 'all',
-        'targets': 'all',
-        'exceptions': True,
-        'rtti': True,
-        'threads': True,
-        'lto': 'Off',
-        'static_stdlib': False,
-        'unwind_tables': True,
-        'expensive_checks': False,
-        'use_perf': False,
-        'use_sanitizer': 'None',
-        'with_ffi': False,
-        'with_zlib': True,
-        'with_xml2': True,
-        'use_llvm_cmake_files': False,
+        "shared": False,
+        "fPIC": True,
+        "components": "all",
+        "exceptions": True,
+        "rtti": True,
+        "threads": True,
+        "lto": "Off",
+        "static_stdlib": False,
+        "unwind_tables": True,
+        "expensive_checks": False,
+        "use_perf": False,
+        "use_sanitizer": "None",
+        "with_libedit": True,
+        "with_ffi": False,
+        "with_terminfo": False,  # differs from LLVM default
+        "with_xml2": True,
+        "with_z3": True,
+        "with_zlib": True,
+        "ram_per_compile_job": "auto",
+        "ram_per_link_job": "auto"
     }
-
-    # Older cmake versions may have issues generating the graphviz output used
-    # to model the components
-    build_requires = [
-        'cmake/3.20.5'
```

Comment:
Why is it not required anymore?

Reply:
Because graphviz is not used to model component dependencies, as noted in the comment.

Reply:
To clarify further - the original recipe used graphviz output which was then parsed and sanitized to generate component dependencies. Instead of doing this I opted to just use the information that LLVM already outputs in its CMake files, I thought this was more robust since the best source of cmake dependencies is cmake itself rather than going via graphviz and back again.